app/controllers/invites_controller.rb

```diff
@@ -109,73 +78,104 @@ def create
 
-    invite_link = if is_single_invite
-      Invite.generate_single_use_invite_link(params[:email], current_user, topic, group_ids)
-    else
-      Invite.generate_multiple_use_invite_link(
-        invited_by: current_user,
+      invite.update!(
         max_redemptions_allowed: params[:max_redemptions_allowed],
-        expires_at: params[:expires_at],
-        group_ids: group_ids
+        expires_at: params[:expires_at]
       )
     end
-    if invite_link.present?
-      render_json_dump(invite_link)
-    else
-      render json: failed_json, status: 422
-    end
-  rescue => e
-    render json: { errors: [e.message] }, status: 422
+
+    render json: success_json
   end
 
   def destroy
     params.require(:id)
 
     invite = Invite.find_by(invited_by_id: current_user.id, id: params[:id])
     raise Discourse::InvalidParameters.new(:id) if invite.blank?
+
     invite.trash!(current_user)
 
     render json: success_json
   end
 
+  def perform_accept_invitation
+    params.require(:id)
+    params.permit(:email, :username, :name, :password, :timezone, user_custom_fields: {})
+
+    invite = Invite.find_by(invite_key: params[:id])
+
+    if invite.present?
+      begin
+        user = invite.redeem(
+          email: invite.is_invite_link? ? params[:email] : invite.email,
+          username: params[:username],
+          name: params[:name],
+          password: params[:password],
+          user_custom_fields: params[:user_custom_fields],
+          ip_address: request.remote_ip
+        )
+
+        if user.present?
+          log_on_user(user) if user.active?
+          user.update_timezone_if_missing(params[:timezone])
+          post_process_invite(user)
+          response = { success: true }
+        else
+          response = { success: false, message: I18n.t('invite.not_found_json') }
+        end
+
+        topic = invite.topics.first
+        if user.present? && user.active?
+          response[:redirect_to] = topic.present? ? path(topic.relative_url) : path("/")
+        elsif user.present?
+          response[:message] = I18n.t('invite.confirm_email')
+          cookies[:destination_url] = path(topic.relative_url) if topic.present?
+        end
+
+        render json: response
+      rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotSaved => e
+        render json: {
+          success: false,
+          errors: e.record&.errors&.to_hash || {},
+          message: I18n.t('invite.error_message')
+        }
+      rescue Invite::UserExists => e
+        render json: { success: false, message: [e.message] }
+      end
+    else
+      render json: { success: false, message: I18n.t('invite.not_found_json') }
+    end
+  end
+
   def rescind_all_invites
     guardian.ensure_can_rescind_all_invites!(current_user)
 
-    Invite.rescind_all_expired_invites_from(current_user)
+    Invite
+      .where(invited_by: current_user)
```

Comment:
Can a staff user rescind the invite of another user? I realized the previous implementation only allows a staff user to rescind invites that were sent out by the user itself.

Reply:
I do not think so and I think this is expected behavior. This was implemented before, so I cannot say for sure.

Reply:
OK :) lets keep it as it is then.